kf_lib_data_ingest/target_api_plugins/kids_first_dataservice.py

```diff
@@ -98,19 +98,27 @@ class Study:
 
     @classmethod
     def get_key_components(cls, record, get_target_id_from_record):
-        return {
-            "data_access_authority": not_none(record[CONCEPT.STUDY.AUTHORITY]),
-            "external_id": not_none(record[CONCEPT.STUDY.ID]),
-        }
+        kfid = record.get(cls.target_id_concept) or record.get(
+            CONCEPT.PROJECT.ID
+        )
+        au = record.get(CONCEPT.STUDY.AUTHORITY)
+        id = record.get(CONCEPT.STUDY.ID)
+        assert (au and id) or kfid
```

Comment:
Referring to the dataservice SD spec (https://kf-api-dataservice.kidsfirstdrc.org/#tag/Study%2Fpaths%2F~1studies%2Fpost), `data_access_authority` is not required?

Reply:
I'm just worried that study external id may not be unique :(

Reply:
ugh, hmm. That is a good point under the assumption that, within the same access authority, a study id should be unique?

Reply:
that's my hope! haha